doc/source/main-documentation/qibolab.rst

```diff
@@ -0,0 +1,611 @@
+Alternatively, you can achieve the same result using the dedicated :class:`qibolab.pulses.DrivePulse` object:
+
+.. code-block:: python
+
+    from qibolab.pulses import *
+
+    pulse = DrivePulse(
+        start=0,  # timing, in all qibolab, is expressed in ns
+        duration=40,
+        amplitude=0.5,  # this amplitude is relative to the range of the instrument
+        frequency=1e8,  # frequency are in Hz
+        relative_phase=0,  # phases are in radians
+        shape=Rectangular(),
+        channel="channel",
+        qubit=0,
+    )
+
+Both the Pulses objects and the PulseShape object have useful plot functions and several different various helper methods.
+
+To organize pulses into sequences, Qibolab provides the :class:`qibolab.pulses.PulseSequence` object. Here's an example of how you can create and manipulate a pulse sequence:
+
+.. code-block:: python
+
+    from qibolab.pulses import PulseSequence
+
+    sequence = PulseSequence()
+
+    sequence.add(pulse1)
+    sequence.add(pulse2)
+    sequence.add(pulse3)
+    sequence.add(pulse4)
+
+    print(f"Total duration: {sequence.duration}")
+
+    sequence_ch1 = sequence.get_channel_pulses("channel1")  # Selecting pulses on channel 1
+    print(f"We have {sequence_ch1.count} pulses on channel 1.")
+
+.. warning::
+
+    Pulses in PulseSequences are ordered automatically following the start time (and the channel if needed). Not by the definition order.
+
+When conducting experiments on quantum hardware, pulse sequences are vital. Assuming you have already initialized a platform, executing an experiment is as simple as:
+
+.. code-block:: python
+
+   result = my_platform.execute_pulse_sequence(sequence)
+
+Lastly, when conducting an experiment, it is not always required to define a pulse from scratch.
+Usual pulses, as pi-pulses or measurements are already defined in the platform runcard and can be easily initialized with platform methods.
+Typical experiments may include both pre-defined pulses and new ones:
+
+.. code-block:: python
+
+    sequence = PulseSequence()
+    sequence.add(my_platform.create_RX_pulse())
+    sequence.add(DrivePulse(...))
+    sequence.add(my_platform.create_MZ_pulse())
+
+    results = my_platform.execute_pulse_sequence(sequence, options=options)
+
+.. note::
+
+   options is an :class:`qibolab.execution_parameters.ExecutionParameters` object, detailed in a separate section.
+
+Symbolic expressions
+--------------------
+
+Sweepers
+--------
+
+Sweeper objects, represented by the :class:`qibolab.sweeper.Sweeper` class, stand as a crucial component in experiments and calibration tasks within the Qibolab framework.
+
+Consider a scenario where a resonator spectroscopy experiment is performed. This process involves a sequence of steps:
+
+1. Define a pulse sequence.
+2. Define a readout pulse with frequency A.
+3. Execute the sequence.
+4. Define a new readout pulse with frequency A + :math:`\epsilon`.
```

Comment:
Minor: these equations look a bit weird because A is just a letter and \epsilon looks like it's compiled to an image. Maybe it would be better to use one of the two approaches.